### ISSUE TITLE
Re-create app-subscriptions after unpublish/publish API

### ISSUE THREAD

**Issue by gwandwingor** [Axway-API-Management-Plus/apim-cli#114]
**API-Mgr. CLI version**
1.3.0

**API-Manager and Service-Pack Version**
7.7 Sept20

**User story**
After an API has been modified by unpublish/publish method, all previously subscribed client applications are not recreated.
The only way to recreate them is to manually add them in the clientApplications of the API config file. 

It would be useful to be able to provide a way to reapply the client Apps subscriptions automatically, especially now that apim-cli is deciding when to unpublish/publish or re-create an API.

**Comment by gwandwingor** [comment]
It seems achievable by doing a few tweaks in the com.axway.apim.apiimport.actions.ManageClientApps class

I did a few simple tests locally wher I removed the exit condition 

`if(desiredState.getApplications()==null && !reCreation) return;`

and set recreateActualApps with actualState apps :

```
if (!reCreation && actualState.getState().equals(API.STATE_PUBLISHED) && CoreParameters.getInstance().getClientAppsMode().equals(CoreParameters.Mode.add)) {
	recreateActualApps = actualState.getApplications();
	createAppSubscription(recreateActualApps, actualState.getId());
	// Update the In-Memory actual state for further processing 
	actualState.setApplications(recreateActualApps);
}
```
It seems to have done the trick for me but there might be other impacts I might not have seen.

**Comment by cwiechmann** [comment]
That is interesting, as the Re-Creation of application subscriptions is part of the existing integration-tests and as the new "mode" is now default, all integration tests running with that mode.

**Comment by gwandwingor** [comment]
It only seems to happen for applications in different organizations than the API.
In the case of apps in the same org, api subscription stays.

**Comment by cwiechmann** [comment]
> It only seems to happen for applications in different organizations than the API.
> In the case of apps in the same org as the API, subscription stays.

Yes, that is expected, as the subscription stays for that applications.

Perhaps, the tests are forcing a Re-Creation of the API as before by using a changed Swagger-Definition. Let me extend one of the Application-Subscription test-cases to have a simple change, like a new a tag to see what happens.

**Comment by cwiechmann** [comment]
I can confirm this issue and I consider this as a major bug, as it may have a big impact for customer losing their current subscriptions. I'm going to fix that now.

**Comment by cwiechmann** [comment]
Quick update: I have already started working it and I came to conclusion, that the decision, which existing applications needs to be re-created should made by the action (e.g. RecreateToUpdateAPI, RepublishToUpdateAPI). The reason is, that only these action classes know the context in which an API is updated, renewed or newly created.
The code in ManageClientApps.java is already quite complicated and difficult to maintain. So that means, work in progress.
